lib/module.js

```diff
@@ -460,22 +458,10 @@ Module._load = function(request, parent, isMain) {
     debug('Module._load REQUEST %s parent: %s', request, parent.id);
   }
 
-  if (isMain && experimentalModules) {
-    (async () => {
-      // loader setup
-      if (!ESMLoader) {
-        ESMLoader = new Loader();
-        const userLoader = process.binding('config').userLoader;
-        if (userLoader) {
-          ESMLoader.isMain = false;
-          const hooks = await ESMLoader.import(userLoader);
-          ESMLoader = new Loader();
-          ESMLoader.hook(hooks);
-        }
-      }
-      Loader.registerImportDynamicallyCallback(ESMLoader);
-      await ESMLoader.import(getURLFromFilePath(request).pathname);
-    })()
+  if (experimentalModules && isMain) {
+    internalESModule.loaderPromise.then((loader) => {
```

Comment:
I'm not a fan of our error handling logic here of `console.error(e); process.exit(1)` but that's unrelated to this PR. I might raise it at the team meeting.

Reply:
@benjamingr that was done because unhandled rejection gave even worse error messages to the user.

Reply:
@bmeck then we should either fix the unhandled rejection error messages or `throw` here in `nextTick` so that people with `uncaughtException` checks can intercept it if they can for synchronous loading?

If you can point me towards related reading I promise to read it :)

Reply:
related reading??

Reply:
For adding the behavior for error messages

Reply:
https://github.com/nodejs/node/pull/14369#issuecomment-317260873